components/aws/sagemaker/tests/integration_tests/component_tests/test_batch_transform_component.py

```diff
@@ -70,7 +70,7 @@ def test_transform_job(
 
     # Verify output location from pipeline matches job output and that the transformed file exists
     output_location = utils.read_from_file_in_tar(
-        output_files["sagemaker-batch-transformation"]["output_location"], "data",
+        output_files["sagemaker-batch-transformation"]["output_location"]
```

Comment:
looks like all output file names are data now, is there a way to set file names as before?

Reply:
Alexey clarified that in a comment above (https://github.com/kubeflow/pipelines/pull/4119#discussion_r448728392). As long as it's one file per output path directory then the default is data. I don't think there's anything wrong with that, though?

Reply:
ok, forgot to remove this, saw his comment later

I guess it is fine for now since we are relying on few defaults - packaged as tar.gz and default file name

Reply:
> As long as it's one file per output path directory then the default is data.

It's more like this: You're given a generated local file path for your output. The actual path is generated. Currently the path seems to end with `data`. If you output a single blob, then you'd want to just write the blob data to the provided path. If you want to output a directory, then you should create a directory at the provided path. In that case, the paths would be `/tmp/outputs/output_name/data/some_file_*.txt`.

>looks like all output file names are data now, is there a way to set file names as before?

Semantically you can divide all output formats into two kinds: blobs and directories. Blobs are nameless. Directories have named files inside. To output a blob, just write to the provided path (e.g. `/tmp/outputs/output_name/data`). To output a directory, create a directory at the provided path (e.g. `/tmp/outputs/output_name/data/`) and write files into it (e.g. `/tmp/outputs/output_name/data/some_file_*.txt`).

Reply:
@Ark-kun Thank you for that explanation. 

1. Is there any reason to create multiple output files (your directory example, for instance)? How would KFP map the output directory to the DSL outputs?

2. At the moment we are serialising everything into raw strings in the files (except for some dicts which we JSON serialise). I think it would be preferable to JSON serialise everything, however when I tried this KFP was passing the raw contents of the file as JSON to the next component (strings were not being unwrapped). Is there a way to specify that the KFP is JSON serialised?